File: leetcode/questions/283/java/283.java
class Solution {
    public void moveZeroes(int[] nums) {
        int nonZero = 0;

        for (int i = 0; i < nums.length; ++i) {
            int aux = nums[i];
            nums[i] = nums[nonZero];
            nums[nonZero] = aux;

            if (nums[nonZero] != 0) {
                nonZero += 1;
            }
        }
    }
}

File: leetcode/questions/283/python/283.py
from typing import List

class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        non_zero = 0

        for idx in range(len(nums)):
            nums[idx], nums[non_zero] = nums[non_zero], nums[idx]
            
            if nums[non_zero] != 0:
                non_zero += 1        